CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -442,34 +497,62 @@ public static function createActivities($form, $html_message, $contactIds) {
     if (!empty($form->_activityId)) {
       $activityParams += array('id' => $form->_activityId);
     }
-    if ($form->_cid) {
-      $activity = CRM_Activity_BAO_Activity::create($activityParams);
-      if (!empty($form->_caseId)) {
-        $caseActivityParams = array('activity_id' => $activity->id, 'case_id' => $form->_caseId);
-        CRM_Case_BAO_Case::processCaseActivity($caseActivityParams);
-      }
+
+    // This seems silly, but the old behavior was to first check `_cid`
+    // and then use the provided `$contactIds`. Probably not even necessary,
+    // but difficult to audit.
+    $contactIds = $form->_cid ? array($form->_cid) : $contactIds;
+
+    $activityContacts = CRM_Core_OptionGroup::values('activity_contacts', FALSE, FALSE, FALSE, NULL, 'name');
```

Comment:
Well, both you and @mickadoo commented about the ugliness of how target-contacts are added, and I owe little cleanup under TCP. 6d3827cb836a3e51a2c5411cc390e04ee60f18b2 removes calls to both  `OptionGroup::values()` and `ActivityContact::create()`.

`<irrelevant-old-complaining>` It's so silly how `Activity::create()` expects pass-by-references.